src/jira/client/jira-client.ts

```diff
@@ -279,6 +291,12 @@ export const getJiraClient = async (
 		},
 		deployment: {
 			submit: async (data, options?: JiraSubmitOptions): Promise<DeploymentsResult> => {
+				// TODO: Looks like gitHubInstallationId is not provided only for
+				// TODO: appState calls. We should split the client in two then, otherwise we are not leveraging TypeScript whatsoever
+				if (!gitHubInstallationId) {
+					throw new Error("Data must belong to repo, and repo must belong to an subscription! This call should be never possible.");
```

Comment:
Wonder if it's worth creating a custom error for this? e.g. something like
```
class WhateverYouWantToCallThisError extends Error {
  constructor(message: string) {
    super(message)
  }
}
```

And maybe storing ""Data must belong to repo, and repo must belong to an subscription! This call should be never possible." to in variable to avoid repetition?

Reply:
I think I'll rip it off in my next PR...